compiler/il/OMRNode.hpp

```diff
@@ -640,10 +645,43 @@ class OMR_EXTENSIBLE Node
 
    TR_YesNoMaybe          hasBeenRun();
 
-   /// Given a monenter node, return the persistent class identifer that's being synchronized
+   /*
+    * The hidden child of a monitor related node is used to store the monitor object's type information.
+    * The type information can be the following two forms:
+    *
+    * 1. The persistent class identifier is stored if the type of a monitor object is a known class.
+    *    Notice java.lang.Object identifier is only stored if it's a fixed class
+    * 2. IDENTITY_CLASS_PLACEHOLDER is stored if the exact monitor type is unknown but is of
+    *    Identity class type
+    */
+   /// Given a monenter node, return the persistent class identifier that's being synchronized
    TR_OpaqueClassBlock *  getMonitorClass(TR_ResolvedMethod *);
+   /*
+    *  \return
+    *    NULL if the concrete class of the monitor object is unknown
+    */
    TR_OpaqueClassBlock *  getMonitorClassInNode();
-   void                   setMonitorClassInNode(TR_OpaqueClassBlock *);
+   /// set the persistent class identifier on the node
+   void                   setMonitorClassInNode(TR_OpaqueClassBlock * classz) { setMonitorInfoInNode(classz); }
```

Comment:
Don't inline definitions need to go in the OMRNode_inlines.hpp for the extensible classes to work properly? @Leonardo2718 I have that right don't I?

Reply:
Inline functions that need to call `self()` must indeed go in `*_inlines.hpp` files, which is true here because it should be `self()->setMonitorInfoInNode(classz);` (see linter failure).